Game/Assets/Scripts/PlayerMovementScript.cpp

```diff
@@ -0,0 +1,71 @@
+#include "PlayerMovementScript.h"
+#include "Engine/SceneBuilder/InputSystem.h"
+#include "Engine/Utilities/Log.h"
+#include "Engine/SceneBuilder/Scene.h"
+#include "Engine/SceneBuilder/InputSystem.h"
+#include "Engine/Scripts/ColorSwap.h"
+
+namespace Engine
+{
+	void PlayerMovementScript::onUpdate(const Engine::DeltaTime& dt)
+	{
+		if (InputSystem::getInstance()->isButtonPressed(0)) {
+
+			float mouseX = InputSystem::getInstance()->getCursorPos().x;
+			float mouseY = InputSystem::getInstance()->getCursorPos().y;
+
+			// Use picking to get entities (with physics component) player clicked on
+			std::list<Entity*> entities = m_entity.getScene()->pick(mouseX, mouseY);
+			//GE_TRACE("Pick @ {0} {1}", mouseX, mouseY);
+			for (auto entity : entities)
+			{
+				//GE_TRACE("Picked this: {0}", entity->getComponent<TagComponent>().tag);
+				if (entity->hasComponent<TransformComponent>())
+				{
+					auto transform = entity->getComponent<TransformComponent>();
+					if (tileInRange(*entity))
+					{
+						static_cast<ColorSwap*>(entity->getComponent<ScriptComponent>().m_instance)->swapMyColor();
+
+						// Move to the tile the player clicked on
+						auto player = m_entity;
+						int x = transform.position.x + 36;
+						int y = transform.position.y - 104;
+						player.getComponent<TransformComponent>().position.x = x;
+						player.getComponent<TransformComponent>().position.y = y;
+					}
+				}
+			}
+		}
+	}
+
+	bool PlayerMovementScript::tileInRange(Entity tile)
+	{
+		// Get the position of the player
+		glm::vec3 playerPos = m_entity.getComponent<TransformComponent>().position;
+		int minX = playerPos.x - 136; // -36 for offset, -100 for the tile
```

Comment:
it would be better to have some of these done programmatically rather than hardcoded

Reply:
Agree.  Personally I think it's ok to leave it like this for now, and if we want/need to change it later, I can use updated script functionality to get the player and tile entities (without having to loop through every entity) and use their values to calculate this programmatically.  Should be pretty easy once we have a "get entity by tag" or something like that